### DISCUSSION TITLE
how can model know the custome class names .

### DISCUSSION THREAD

**Discussion by yvr354** [ultralytics/ultralytics#8707] [Q&A]
how can model know the custome class names . even i dint gave any data yaml files . ca any one explain me .
from ultralytics import YOLO

# Load a pretrained YOLOv8n model
model = YOLO(f"diff_size/weights/best.pt")

# Run inference on 'bus.jpg' with arguments
model.predict(f"doubt_orginal", save=True,  conf=0.6)   . here i am not giving any data yaml file .

**Comment by pderrenger** [comment]
Hello! 😊 When you train a YOLOv8 model on a custom dataset, the class names are learned during the training process and are embedded within the model's `.pt` file. This means that when you load your trained model using `YOLO(f"diff_size/weights/best.pt")`, it already knows the class names from training, even if you don't explicitly provide a `.yaml` file during inference.

Here's a quick example of how it works:

```python
from ultralytics import YOLO

# Load your custom trained YOLOv8n model
model = YOLO("diff_size/weights/best.pt")

# Run inference on an image without specifying class names
results = model.predict("path/to/your/image.jpg", save=True, conf=0.6)

# The model uses the class names it learned during training
```

So, as long as your model was properly trained with the custom class names, it will automatically use them during inference. No need to worry about providing the class names again! 🚀
